### PR TITLE
SAML Portal: Support query and mutate saml secrets

### PR DESCRIPTION
**Pull request by tung2744**
ref DEV-1706

TODO:

- [x] Update tests

### CODE REVIEW

File: pkg/lib/config/secret_update_instruction.go [hunk 1]
```diff
@@ -569,9 +577,66 @@ func (i *BotProtectionProviderCredentialsUpdateInstruction) unset(currentConfig
...
+	}
+
+	if credentials == nil {
+		credentials = &SAMLIdpSigningMaterials{
+			Certificates: []*SAMLIdpSigningCertificate{},
+		}
+	}
+
+	// FIXME: The common name is not important, but we might want to put the app id here.
+	newCert, err := GenerateSAMLIdpSigningCertificate("urn:authgear.com")
```
Comment (louischan-oursky) [inline pkg/lib/config/secret_update_instruction.go]: What value did we use previously?

File: pkg/lib/config/configsource/resources.go [hunk 2]
```diff
@@ -538,6 +538,10 @@ func (d AuthgearSecretYAMLDescriptor) UpdateResource(ctx context.Context, _ []re
 		// Since the key neither use for sig nor enc
 		GenerateClientSecretOctetKeyFunc: secrets.GenerateOctetKey,
 		GenerateAdminAPIAuthKeyFunc:      secrets.GenerateRSAKey,
+		GenerateSAMLIdpSigningCertificate: func() (*config.SAMLIdpSigningCertificate, error) {
+			// FIXME: The common name is not important, but we might want to put the app id here.
```
Comment (louischan-oursky) [inline pkg/lib/config/configsource/resources.go:542]: What value does other competitor put in here?

File: pkg/lib/config/secret_update_instruction.go (hunk 1 above)
Comment (tung2744) [reply pkg/lib/config/secret_update_instruction.go]: For existing projects, I created the certs manually with a value like `urn:project_id.authgear.com`

File: pkg/lib/config/configsource/resources.go (hunk 2 above)
Comment (tung2744) [reply pkg/lib/config/configsource/resources.go:542]: It is something like an id.

Comment (louischan-oursky) [reply pkg/lib/config/configsource/resources.go:542]: I suggest using `APP_HOST_SUFFIX` (which is something like `.authgear.cloud`) to derive the common name here.

Comment (tung2744) [reply pkg/lib/config/configsource/resources.go]: Updated to use the saml entity id here.

**Review by louischan-oursky — APPROVED** [review]

**Pull request merged**